src/Annotations/Assertion.php

```diff
@@ -21,26 +22,34 @@
  *   @Attribute("constraint", type = "Symfony\Component\Validator\Constraint[]|Symfony\Component\Validator\Constraint")
  * })
  */
+#[Attribute(Attribute::TARGET_METHOD)]
 class Assertion implements ParameterAnnotationInterface
 {
-    /** @var string */
-    private $for;
+    private string $for;
     /** @var Constraint[] */
-    private $constraint;
+    private array $constraint;
 
-    /** @param array<string, mixed> $values */
-    public function __construct(array $values)
-    {
-        if (! isset($values['for'])) {
-            throw new BadMethodCallException('The @Assert annotation must be passed a target. For instance: "@Assert(for="$email", constraint=@Email)"');
+    /**
```

Comment:
This looks like a breaking change, no ? I mean people upgrading might get errors if they are using Annotations.

Reply:
yes that's BC to support symfony 7